routes/user.js

```diff
@@ -13,35 +13,42 @@ exports.create = function(req, res){
 	var description = req.body.description;
 
 	// Check that the provided information is valid.
-	if (!validator.isEmail(email)){
-		return resError(res, "Please enter a valid email address as your username.");
+	if (!validator.isEmail(email) || !validator.normalizeEmail(email)){
+		return resError(res, "INVALID_EMAIL");
 	} else if (!validator.isLength(firstName, 2, 128) || !validator.isLength(lastName, 2, 128)){
-		return resError(res, "Sorry, your name is too short or too long.  " +
-				"Do you have a nickname between 2 and 128 characters?");
+		return resError(res, "BAD_NAME_LENGTH");
 	}
 
-	// Hash the password.
-	var passHash = bcrypt.hashSync(req.body.password, salt);
+	// Check for another pre-existing user with the same email.
+	User.find({ username: validator.escape(email) }, 'username', function(err, docs){
+		// Stop the request if there was a DB error of it another user has the same username.
```

Comment:
I think this comment may have a grammatical error in it. Is the `of it` part correct in the middle or is it meant to be something else?